scripts/create-secrets/k8sExecution.js

```diff
@@ -0,0 +1,320 @@
+    });
+}
+
+function getEnvByClusterType(config) {
+    const localClusterType = config.get("local-cluster-type");
+    if (
+        typeof localClusterType === "undefined" ||
+        localClusterType !== "minikube"
+    ) {
+        return Object.assign({}, process.env);
+    }
+    const dockerEnvProcess = childProcess.execSync(
+        "minikube docker-env --shell bash",
+        { encoding: "utf8" }
+    );
+    const dockerEnv = dockerEnvProcess
+        .split("\n")
+        .filter(line => line.indexOf("export ") === 0)
+        .reduce(function(env, line) {
+            const match = /^export (\w+)="(.*)"$/.exec(line);
+            if (match) {
+                env[match[1]] = match[2];
+            }
+            return env;
+        }, {});
+
+    const env = Object.assign({}, process.env, dockerEnv);
+    return env;
+}
+
+function overrideSettingWithEnvVars(configData) {
+    getEnvVarInfo().forEach(item => {
+        const envVal = env[item.name];
+        if (typeof envVal === "undefined") return;
+        configData[item.settingName] = envVal;
+    });
+    if (
+        configData["use-regcred-password-from-env"] === true &&
+        env["CI_JOB_TOKEN"] &&
+        !env[settingNameToEnvVarName("regcred-password")]
+    ) {
+        configData["regcred-password"] = env["CI_JOB_TOKEN"];
+    }
+    if (
+        configData["get-namespace-from-env"] === true &&
+        env["CI_COMMIT_REF_SLUG"] &&
+        !env[settingNameToEnvVarName("cluster-namespace")]
+    ) {
+        configData["cluster-namespace"] = env["CI_COMMIT_REF_SLUG"];
+    }
+    if (
+        typeof configData["manual-db-passwords"] === "object" &&
+        configData["manual-db-passwords"]["answer"] === false &&
+        configData["manual-db-passwords"]["password"] &&
+        !env[settingNameToEnvVarName("db-passwords")]
+    ) {
+        configData["db-passwords"] =
+            configData["manual-db-passwords"]["password"];
+    }
+    return configData;
+}
+
+function validKubectl() {
+    try {
+        childProcess.execSync("kubectl", {
+            stdio: "ignore",
+            env: env
+        });
+    } catch (e) {
+        throw new Error(
+            `Failed to execute \`kubectl\` utility: ${e}\n` +
+                "Make sure you have install & config `kubectl` properly before try again."
+        );
+    }
+}
+
+function checkNamespace(namespace) {
+    try {
+        childProcess.execSync(`kubectl get namespace ${namespace}`, {
+            stdio: "ignore",
+            env: env
+        });
+        return true;
+    } catch (e) {
+        console.log(
+            chalk.red(
+                `Failed to get k8s namespace ${namespace} or namespace has not been created yet: ${e}`
+            )
+        );
+        return false;
+    }
+}
+
+function createNamespace(namespace) {
+    childProcess.execSync(`kubectl create namespace ${namespace}`, {
+        stdio: "inherit",
+        env: env
+    });
+}
+
+function getTplObj(name, namespace) {
```

Comment:
I assume this is to get a template object? Can we call it `buildTemplateObject` ?

Reply:
👍